### PR TITLE
Add lambda to create db users

### PR DESCRIPTION
**Pull request by MancunianSam**
Also, turn on iam auth for the consignment db.

### CODE REVIEW

File: modules/consignment-api/outputs.tf [hunk 1]
```diff
@@ -17,4 +17,8 @@ output "database_password" {
 
 output "api_url" {
   value = "https://${aws_route53_record.api_dns.name}.${var.dns_zone_name_trimmed}"
+}
+
+output "consignment_db_security_group_id" {
+  value = aws_security_group.database.id
 }
```
Comment (TomJKing) [inline modules/consignment-api/outputs.tf:24]: Missing line break

File: modules/consignment-api/variables.tf [hunk 2]
```diff
@@ -37,4 +37,8 @@ variable "dns_zone_name_trimmed" {}
 variable "ip_whitelist" {
   description = "IP addresses allowed to access"
   default     = ["0.0.0.0/0"]
+}
+
+variable "create_users_security_group" {
+  type = list(string)
 }
```
Comment (TomJKing) [inline modules/consignment-api/variables.tf:44]: Missing line break

File: modules/consignment-api/variables.tf [hunk 3]
```diff
@@ -37,4 +37,8 @@ variable "dns_zone_name_trimmed" {}
 variable "ip_whitelist" {
   description = "IP addresses allowed to access"
   default     = ["0.0.0.0/0"]
+}
+
+variable "create_users_security_group" {
```
Comment (TomJKing) [inline modules/consignment-api/variables.tf:42]: Maybe add `_id` onto the end of the variable name to make clear an id value

File: root.tf [hunk 4]
```diff
@@ -263,6 +264,20 @@ module "checksum_lambda" {
...
+  project                    = var.project
+  common_tags                = local.common_tags
+  lambda_create_db_users     = true
+  vpc_id                     = module.shared_vpc.vpc_id
+  private_subnet_ids         = module.backend_checks_efs.private_subnets
+  consignment_database_sg_id = module.consignment_api.consignment_db_security_group_id
+  db_admin_user              = module.consignment_api.database_username
+  db_admin_password          = module.consignment_api.database_password
+  db_url                     = module.consignment_api.database_url
+
```
Comment (TomJKing) [inline root.tf:278]: Unnecessary line break

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments. Assume this requires PR: https://github.com/nationalarchives/tdr-terraform-modules/pull/86 to be deployed first.

File: modules/consignment-api/variables.tf (hunk 3 above)
Comment (MancunianSam) [reply modules/consignment-api/variables.tf:42]: Yep,  good point, I've changed it.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**